### PR TITLE
ref(tagstore): Support more model import strategies

### PR DESCRIPTION
**Pull request by tkaemming**

### CODE REVIEW

File: src/sentry/tagstore/models.py [hunk 1]
```diff
@@ -11,23 +11,46 @@
...
+    backends = [
+        backend: backend['path'] for backend in
+        settings.SENTRY_TAGSTORE_OPTIONS.get('backends', {}).values()
+    ]
 elif settings.SENTRY_TAGSTORE.startswith('sentry.tagstore.multi'):
-    for backend in settings.SENTRY_TAGSTORE_OPTIONS.get('backends', []):
-        if backend[0].startswith('sentry.tagstore.legacy'):
-            from sentry.tagstore.legacy.models import *  # NOQA
-        elif backend[0].startswith('sentry.tagstore.v2'):
-            from sentry.tagstore.v2 import models as _v2_models  # NOQA
```
Comment (tkaemming) [inline src/sentry/tagstore/models.py]: This import order is now non-deterministic, if we want to prioritize one over the other we can sort them by some arbitrary score.

File: src/sentry/tagstore/models.py [hunk 2]
```diff
@@ -11,23 +11,46 @@
...
+            if i == 0:
+                # If this is the first iteration of the loop, we need to
+                # emulate ``from x import *`` by copying the module contents
+                # into the local (module) scope. This follows the same rules as
+                # the import statement itself, as defined in the refrence docs:
+                # https://docs.python.org/2.7/reference/simple_stmts.html#import
+                if getattr(models, '__all__'):
+                    predicate = lambda name: name in models.__all__
+                else:
+                    predicate = lambda name: not name.startswith('_')
```
Comment (tkaemming) [inline src/sentry/tagstore/models.py]: God help me if the linter chokes on this like it was trying to do locally

**Review by tkaemming — COMMENTED** [review]
This is mostly to support `ServiceDelegator`.

File: src/sentry/tagstore/models.py (hunk 2 above)
Comment (bretthoerner) [reply src/sentry/tagstore/models.py]: `'_'`

Comment (MaxBittker) [reply src/sentry/tagstore/models.py]: = |

Comment (mattrobenolt) [reply src/sentry/tagstore/models.py]: Hi everyone.

File: src/sentry/tagstore/models.py [hunk 3]
```diff
@@ -11,23 +11,46 @@
...
+                locals().update({k: v for k, v in vars(models).items() if predicate(k)})
+            break
+    else:
+        raise ImproperlyConfigured("Found unknown tagstore backend '%s'" % backend)
```
Comment (bretthoerner) [inline src/sentry/tagstore/models.py]: I'm confused about the previous 3 lines. What's the `break` about and that `else` is on the `for` (not the `if`) right?

**Review by bretthoerner — APPROVED** [review]
Syntax b00g otherwise makes sense. I think. :face_with_head_bandage:

File: src/sentry/tagstore/models.py [hunk 4]
```diff
@@ -11,23 +11,46 @@
...
-elif settings.SENTRY_TAGSTORE.startswith('sentry.tagstore.v2'):
-    from sentry.tagstore.v2.models import *  # NOQA
-
-    # Prevent schema migration from adding DROP TABLE for Legacy
-    from sentry.tagstore.legacy import models as _legacy_models  # NOQA
+backends = []
+
+if settings.SENTRY_TAGSTORE == 'sentry.utils.services.ServiceDelegator':
+    backends = [
+        backend: backend['path'] for backend in
```
[the remark this answers is not in the document]
Comment (tkaemming) [reply src/sentry/tagstore/models.py]: Shit

**Pull request merged**